File: garden/seeds/java/app/src/test/java/com/WATER_ME_WITH_COMPANY/android/WATER_ME_WITH_APP_NAME/activity/RxActivityTest.java
package com.{{company_name}}.android.{{app_package_name_prefix}}.activity;

import com.{{company_name}}.android.{{app_package_name_prefix}}.{{app_class_prefix}}TestRunner;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.util.ActivityController;

import com.trello.rxlifecycle.ActivityEvent;
import rx.observers.TestSubscriber;

@RunWith({{app_class_prefix}}TestRunner.class)
public class RxActivityTest {

    @Test
    public void testLifecycleObservable() {
        ActivityController<LameRxActivity> controller = Robolectric.buildActivity(LameRxActivity.class);

        final TestSubscriber<ActivityEvent> subscriber = new TestSubscriber<>();

        controller.get().lifecycle().subscribe(subscriber);

        controller.create()
                .start()
                .resume()
                .pause()
                .stop()
                .destroy();

        subscriber.assertValues(
                ActivityEvent.CREATE,
                ActivityEvent.START,
                ActivityEvent.RESUME,
                ActivityEvent.PAUSE,
                ActivityEvent.STOP,
                ActivityEvent.DESTROY
        );
    }

    /**
     * RxActivity is an abstract class, so we need a concrete implementation here..
     */
    public static class LameRxActivity extends RxActivity {

    }
}

File: garden/seeds/kotlin/app/src/test/kotlin/com/WATER_ME_WITH_COMPANY/android/WATER_ME_WITH_APP_NAME/activity/RxActivityTest.kt
package com.{{company_name}}.android.{{app_package_name_prefix}}.activity

import com.{{company_name}}.android.{{app_package_name_prefix}}.{{app_class_prefix}}TestRunner
import com.trello.rxlifecycle.ActivityEvent

import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.Robolectric
import org.robolectric.util.ActivityController

import rx.observers.TestSubscriber

RunWith({{app_class_prefix}}TestRunner::class)
public class RxActivityTest {

    @Test
    public fun testLifecycleObservable() {
        val controller = Robolectric.buildActivity(javaClass<LameRxActivity>())

        val subscriber = TestSubscriber<ActivityEvent>()

        controller.get().lifecycle().subscribe(subscriber)

        controller.create().start().resume().pause().stop().destroy()

        subscriber.assertValues(
                ActivityEvent.CREATE,
                ActivityEvent.START,
                ActivityEvent.RESUME,
                ActivityEvent.PAUSE,
                ActivityEvent.STOP,
                ActivityEvent.DESTROY
        )
    }

    /**
     * RxActivity is an abstract class, so we need a concrete implementation here..
     */
    public class LameRxActivity : RxActivity()
}